static/index.js

```diff
@@ -3,16 +3,29 @@
   // when using `NativeCompileCache` for the first time.
   require('cached-run-in-this-context')
 
+  // Define the window start time before the requires so we get a more accurate
+  // window:start marker.
+  const startWindowTime = Date.now()
+
   const electron = require('electron')
   const path = require('path')
   const Module = require('module')
   const getWindowLoadSettings = require('../src/get-window-load-settings')
+  const StartupTime = require('../src/startup-time')
   const entryPointDirPath = __dirname
   let blobStore = null
   let useSnapshot = false
 
+  const startupMarkers = electron.remote.getCurrentWindow().startupMarkers
```

Comment:
Have you profiled the overhead of this? Under the hood it does a synchronous RPC and might cost us a few milliseconds. On the other hand I don't have a great idea for an alternative approach other than finding some async approach to perform the RPC.

Reply:
Good point! Just measured it with `performance.now()` and that call takes 3.2-3.5ms (the time is quite consistent on my machine).

If we consider that this is too much, we can always add some sampling to e.g only log the markers data on 1/10 of the starts.

Reply:
That feels like an unfortunate amount of time. Is there any way we can fetch this asynchronously?

Reply:
I'm not sure, we're following the same pattern in a few other places (like getting [all the loadSettings](https://github.com/atom/atom/blob/master/src/get-window-load-settings.js#L7)) so I thought it was ok.

I'm gonna check if I can speed this up somehow... if not we can just sample it

Reply:
Ok I took more measures and compared the time this call takes to [the call to load the settings](all the loadSettings](https://github.com/atom/atom/blob/master/src/get-window-load-settings.js#L7), which happens a few lines below this, and turns out that these 3ms may include some bootstrap time for the IPC or something... let me show you the timings:

**With this call**:

- `getTimingSettings`: 3.5ms
- `getWindowLoadSettings`: 1.5ms

**Removing this call**:

- `getWindowLoadSettings`: 4ms

So basically, the net time that's added by this call is ~1ms

Reply:
Okay, I think it's reasonable to just proceed for now. Measuring this will allow us to attack other aspects and we'll probably get the time back and then some.